backend/src/pdf_gen/models/mod.rs

```diff
@@ -0,0 +1,55 @@
+mod model_na_31_2;
+
+use std::path::PathBuf;
+
+pub use model_na_31_2::*;
+use typst::foundations::Bytes;
+
+/// Defines the available models and what their input parameters are.
+pub enum PdfModel {
+    ModelNa31_2(ModelNa31_2Input),
+}
+
+impl PdfModel {
+    /// Get the filename for the input and template
+    pub fn as_filename(&self) -> &'static str {
+        match self {
+            Self::ModelNa31_2(_) => "model-na-31-2",
+        }
+    }
+
+    /// Get the path for the template of this model
+    pub fn as_template_path(&self) -> PathBuf {
+        let mut pb: PathBuf = [self.as_filename()].iter().collect();
+        pb.set_extension("typ");
+
+        pb
+    }
+
+    /// Get the path for the input of this model
+    pub fn as_input_path(&self) -> PathBuf {
+        let mut pb: PathBuf = ["inputs", self.as_filename()].iter().collect();
+        pb.set_extension("json");
+
+        pb
+    }
+
+    /// Get the input, serialized as json
+    pub fn get_input(&self) -> serde_json::Result<Bytes> {
+        let data = match self {
+            Self::ModelNa31_2(input) => serde_json::to_string(input),
+        }?;
+
+        Ok(Bytes::from(data.as_bytes()))
+    }
+
+    pub fn from_name_with_input(
+        name: &str,
+        input: &str,
+    ) -> Result<PdfModel, Box<dyn std::error::Error>> {
+        match name {
+            "model-na-31-2" => Ok(Self::ModelNa31_2(serde_json::from_str(input)?)),
+            _ => Err(std::io::Error::new(std::io::ErrorKind::InvalidInput, "Unknown model").into()),
```

Comment:
Both `std::error::Error` and `std::io::Error` are used, so importing them as `Error` won't work. I think could import `std::io` so that we can write `io::Error` instead.

Reply:
That would be fine, and then importing InvalidInput fully at the start of the file would also help

Reply:
I've used a local use statement to allow importing `std::io::Error` within the scope of this function, while importing `std::error::Error` outside the scope that makes it much more readable.